Assets/Scripts/MoveToClickPoint.cs

```diff
@@ -0,0 +1,56 @@
+// MoveToClickPoint.cs
+using UnityEngine;
+using UnityEngine.AI;
+    
+public class MoveToClickPoint : MonoBehaviour
+{
+    [SerializeField] private NavMeshAgent player;
+    [SerializeField] private Animator anim;
+    [SerializeField] private float distanceToTarget = 2f;
```

Comment:
Perhaps `distanceThreshold` or `destinationDistanceThreshold` would be more descriptive, `distanceToTarget` makes it sound like it holds the same information as "between" does.